### ISSUE TITLE
truffle-contract@4.0.1 - React app - "ethers" -  Module not found: Can't resolve 'xmlhttprequest'

### ISSUE THREAD

**Issue by gjeanmart** [ConsenSys-archive/truffle#1614]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

Importing `truffle-contract@4.0.1` in a React app breaks the web-app compilation

```
Failed to compile.

./node_modules/ethers/utils/web.js
Module not found: Can't resolve 'xmlhttprequest' in '/home/gjeanmart/workspace/sandbox/myapp-react-truffle-contract-issue/node_modules/ethers/utils'
```

## Steps to Reproduce
1. Create a new React app calle `myapp`
```
$ npx create-react-app myapp
```

2. Go to the project folder 
```
$ cd myapp
```

3. Install` truffle-contract` version `4.0.1` as a dependency
```
$ npm install truffle-contract@4.0.1 --save
```

4. Edit `src/App.js` to import `truffle-contract`
```
$ vi src/App.js
import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import TruffleContract from 'truffle-contract'; // <<<<<<<<<<<<<<<<<<<< IMPORT BREAKING THE BUILD

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

export default App;
```

5. Start the React App
```
$ npm start
```

## Expected Behavior

Should start the app nicely on http://localhost:3000/

```
Compiled successfully!

You can now view react-truffle-contract-4.0.1 in the browser.

  Local:            http://localhost:3000/
  On Your Network:  http://192.168.0.59:3000/

Note that the development build is not optimized.
To create a production build, use npm run build.


```

## Actual Results

The compilation fails and throws the following exception:
```
Failed to compile.

./node_modules/ethers/utils/web.js
Module not found: Can't resolve 'xmlhttprequest' in '/home/gjeanmart/workspace/sandbox/myapp-react-truffle-contract-issue/node_modules/ethers/utils'
```

## Environment

* Operating System: Ubuntu 18
* Ethereum client: Ganache
* Truffle version (`truffle version`): 5.0.1
* node version (`node --version`): v8.10.0
* npm version (`npm --version`):  6.1.0

**Comment by juharris** [comment]
I'm getting the same issues after bumping the truffle-contract and react versions.
Your note about importing truffle-contract@4.0.1 gave me an idea:
There's a React Truffle example here: https://github.com/truffle-box/react-box
which doesn't import truffle-contract.  I tested it out and it works for me. I'm going to make my project more like that one and hopefully I won't keep getting issues.

**Comment by juharris** [comment]
So I got my project working by following what I said just above.
I then had to change my code to use the contract interface provided by web3: https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html

**Comment by filips123** [comment]
I also have similar issue (#1617) but without React (just Webpack).

**Comment by adnan214** [comment]
I was able to get around it by upping `ethers` to `v4.0.20`. 

`npm install ethers@4.0.20`

`ethers` `v4.0.0-beta.1` gets installed as a dependency of `web3-eth-abi` `v1.0.0-beta.37`

Still testing but SFSG.

**Comment by filips123** [comment]
Maybe this should be reported to `web3-eth-abi`.

**Comment by filips123** [comment]
This is now fixed in `web3@1.0.0-beta.38`. This package should now also be updated.

**Comment by gjeanmart** [comment]
Thanks!
